prime-router/src/main/kotlin/azure/BlobAccess.kt

```diff
@@ -155,19 +180,23 @@ class BlobAccess() : Logging {
         }
 
         /** Checks if a blob actually exists in the blobstore */
-        fun exists(blobUrl: String): Boolean {
-            return getBlobClient(blobUrl).exists()
+        fun exists(blobUrl: String, blobConnInfo: BlobContainerMetadata = defaultBlobMetadata): Boolean {
+            return getBlobClient(blobUrl, blobConnInfo).exists()
         }
 
         /**
          * Download the blob at the given [blobUrl] as a ByteArray
          */
-        fun downloadBlobAsByteArray(blobUrl: String, retries: Int = getBlobDownloadRetry()): ByteArray {
+        fun downloadBlobAsByteArray(
+            blobUrl: String,
+            blobConnInfo: BlobContainerMetadata = defaultBlobMetadata,
+            retries: Int = blobDownloadRetryCount
```

Comment:
Specifying a function as the default value of `retries` caused some strange mocking behavior. As we don't expect the default number of retries to vary at runtime, I moved the retrieval of this value to private values in the companion object instead, so that a static value is available when this is not provided by the caller.